tools/cmd/scadctl/cmd/gardener.go

```diff
@@ -16,32 +43,512 @@ var scenarioDir string
+		return nil, err
+	}
+
+	return &access{
+		shootCoord:    shootCoords,
+		scheme:        clientScheme,
+		shootClient:   shootClient,
+		controlClient: controlClient,
+	}, nil
+}
+
+func registerSchemes() *runtime.Scheme {
+	scheme := runtime.NewScheme()
+	utilruntime.Must(corev1.AddToScheme(scheme))
+	utilruntime.Must(nodev1.AddToScheme(scheme))
+	utilruntime.Must(schedulingv1.AddToScheme(scheme))
+	utilruntime.Must(storagev1.AddToScheme(scheme))
+	return scheme
+}
+
+func getClient(ctx context.Context, shootCoord ShootCoordinate, scheme *runtime.Scheme, plane GardenerPlane) (client.Client, error) {
+	var targetStr string
+
+	switch plane {
+	case VirtualGardenPlane:
+		targetStr = fmt.Sprintf("gardenctl target --garden %s", shootCoord.Landscape)
+	case DataPlane:
+		targetStr = fmt.Sprintf("gardenctl target --garden %s --project %s --shoot %s",
+			shootCoord.Landscape, shootCoord.Project, shootCoord.Shoot)
+	case ControlPlane:
+		targetStr = fmt.Sprintf("gardenctl target --garden %s --project %s --shoot %s --control-plane",
+			shootCoord.Landscape, shootCoord.Project, shootCoord.Shoot)
+	default:
+		return nil, fmt.Errorf("unsupported gardener plane: %d", plane)
+	}
+
+	cmdStr := fmt.Sprintf("eval $(gardenctl kubectl-env bash) && %s > /dev/null && gardenctl kubectl-env bash", targetStr)
+	cmd := exec.CommandContext(ctx, "bash", "-c", cmdStr)
+	cmd.Env = append(os.Environ(), "GCTL_SESSION_ID=dev")
+
+	capturedOut, err := invokeCommand(cmd)
+	if err != nil {
+		return nil, fmt.Errorf("failed to execute gardenctl command: %w", err)
+	}
+
+	kubeConfigPath, err := extractKubeConfigPath(capturedOut)
+	if err != nil {
+		return nil, err
+	}
+
+	restCfg, err := clientcmd.BuildConfigFromFlags("", kubeConfigPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create rest.Config from kubeconfig %q: %w", kubeConfigPath, err)
+	}
+
+	return client.New(restCfg, client.Options{Scheme: scheme})
+}
+
+// ---------------------------------------------------------------------------------
+// Cluster Snapshot
+// ---------------------------------------------------------------------------------
+func (a *access) ListNodes(ctx context.Context, criteria mkapi.MatchCriteria) (nodes []corev1.Node, err error) {
+	var nodeList corev1.NodeList
+	err = a.shootClient.List(ctx, &nodeList, &client.ListOptions{
+		LabelSelector: criteria.LabelSelector,
+	})
+	if err != nil {
+		return nil, err
+	}
+	if criteria.Names.Len() <= 0 {
+		return nodeList.Items, nil
+	}
+	for _, n := range nodeList.Items {
+		if criteria.Names.Has(n.Name) {
+			nodes = append(nodes, n)
+		}
+	}
+	return nodes, err
+}
+
+func (a *access) ListPods(ctx context.Context, criteria mkapi.MatchCriteria) (pods []corev1.Pod, err error) {
+	var podList corev1.PodList
+	err = a.shootClient.List(ctx, &podList, &client.ListOptions{
+		Namespace:     criteria.Namespace,
+		LabelSelector: criteria.LabelSelector,
+	})
+	if err != nil {
+		return nil, err
+	}
+	if criteria.Names.Len() <= 0 {
```

Comment:
if there are pod names are not specified then you choose not to filter out `schedule gated pods` but if these are then you do?